packages/clarity-js/src/data/metadata.ts

```diff
@@ -93,10 +91,9 @@ export function metadata(cb: MetadataCallback, wait: boolean = true): void {
   if (data && wait === false) {
     // Immediately invoke the callback if the caller explicitly doesn't want to wait for the upgrade confirmation
     cb(data, !config.lean);
-  } else {
-    // Save the callback for future reference; so we can inform the caller when page gets upgraded and we have a valid playback flag
-    callback = cb;
   }
+  
+  callback.push({callback: cb, wait: wait });
```

Comment:
Wouldn't this invoke the call back twice when you execute the if condition? Shouldn't the if() case return without any queuing?